### ISSUE TITLE
SSL Hosts

### ISSUE THREAD

**Issue by zmjwong** [engintron/engintron#63]
Hello; We started testing Engintron and love the work you have put in so far.

We noticed that nginx doesn't bind to port 443 and none of the SSL hosts are being cached by nginx and due to the non binding apache is handling all connections.

Due to the advancement of LetsEncrypt and Googles push to SSL hosts gaining a better SEO; the industry is moving to SSL only connections.

We would love to see some integration in Engintron for importing WHM SSL Hosts into the configuration and adding the configurations in the server field.

**Comment by fevangelou** [comment]
This was an intentional behaviour for this new version (1.5.x) as it handles the whole reverse caching proxy concept differently to other plugins. In fact, it's a lot more efficient to do both (reverse proxying and caching). The caveat (for now) is SSL. Once quick solution is to use a multi-domain SSL certificate (also called UCC/SAN), so using that is probably the simplest way as you'd just have to copy the port 80 block from default.conf and place it lower for the 443 port, plus the SSL specifics. Then you'd just switch Apache SSL to another port (e.g. 8443) and have Nginx SSL proxy to Apache non-SSL.

Let's Encrypt is still in its early steps and the process to generate certificates is a bit cumbersome for now (but nothing that couldn't be handled with smart configurations). Here they state that UCC/SAN certificates will be allowed to issue when they officially launch (they are currently in beta state): https://github.com/letsencrypt/letsencrypt/issues/66

So if this is soon enough, it'll definitely allow for a "cleaner" solution with Engintron and just 1 SSL definition. Not to mention the benefit of being able to use one shared IP.

As a sidenote, one of the dozens of servers we've already deployed Engintron on already uses CloudFlare's flexible SSL option to offer SSL on all their domains by proxying to Nginx's HTTP port. This is also a good and working solution, sans the hassle to move all your domain's DNS to CF.

To summarize, SSL is definitely on our roadmap as we've already included it since the release of v1.5.0: https://github.com/nuevvo/engintron/wiki/Engintron-Roadmap

If Let's Encrypt goes out of beta soon, we'll have Engintron updated in a few days to support UCC/SAN SSL certificates :)

**Comment by imagina** [comment]
Let's Encrypt would be great with an option to load a custom SSL because some of our customers prefer to have a specific brand SSL. Comodo, GeoTrust, etc

**Comment by fevangelou** [comment]
@imagina Of course, having a few exclusions is no big deal.

Since this is part of the roadmap now, I'm closing the issue for now.
